app/src/org/commcare/android/util/MediaUtil.java

```diff
@@ -1,82 +1,274 @@
 package org.commcare.android.util;
 
+import android.content.Context;
 import android.graphics.Bitmap;
 import android.graphics.BitmapFactory;
+import android.util.DisplayMetrics;
+import android.util.Log;
+import android.util.Pair;
+import android.view.Display;
+import android.view.WindowManager;
 
-import org.commcare.android.javarosa.AndroidLogger;
-import org.javarosa.core.model.data.GeoPointData;
-import org.javarosa.core.model.data.UncastData;
+import org.commcare.dalvik.preferences.DeveloperPreferences;
 import org.javarosa.core.reference.InvalidReferenceException;
-import org.javarosa.core.reference.Reference;
 import org.javarosa.core.reference.ReferenceManager;
-import org.javarosa.core.services.Logger;
 
-import java.io.IOException;
+import java.io.File;
 
 /**
  * @author ctsims
  */
 public class MediaUtil {
+
     public static final String FORM_VIDEO = "video";
     public static final String FORM_AUDIO = "audio";
     public static final String FORM_IMAGE = "image";
-	public static Bitmap getScaledImageFromReference(String jrReference) {
-		return getScaledImageFromReference(jrReference, 1);
-	}
-	
-	/*
-	 * returns the file referenced by jrReference scaled by scaleFactor. scaleFactor should be a power of two;
-	 * for example, a scale factor of 4 returns an image that is 1/4 the width/height of the original, and 1/16th
-	 * the size in pixels
-	*/
-	public static Bitmap getScaledImageFromReference(String jrReference, int scaleFactor) {
-        //TODO: Eventually we'll want to be able to deal with dymanic resources here.
-        try {
 
-        Reference imageRef = ReferenceManager._().DeriveReference(jrReference);
-        if(!imageRef.doesBinaryExist()) {
+    /**
+     * Attempts to inflate an image from a CommCare UI definition source.
+     *
+     * @param jrUri The image to inflate
+     * @param boundingWidth the width of the container this image is being inflated into, to serve
+     *                      as a max width. If passed in as -1, gets set to screen width
+     * @param boundingHeight the height fo the container this image is being inflated into, to
+     *                       serve as a max height. If passed in as -1, gets set to screen height
+     * @return A bitmap if one could be created. Null if error occurs or the image is unavailable.
+     */
+    public static Bitmap inflateDisplayImage(Context context, String jrUri,
+                                             int boundingWidth, int boundingHeight) {
+        if (jrUri == null || jrUri.equals("")) {
             return null;
         }
-        
-        BitmapFactory.Options options = new BitmapFactory.Options();
+        try {
+            String imageFilename = ReferenceManager._().DeriveReference(jrUri).getLocalURI();

```

Comment:
This used to work with jr refs that didn't point to files, we should probably have a fallback for what to do here when this doesn't point to a file